server/src/test/java/org/candlepin/policy/AutobindRulesTest.java

```diff
@@ -775,6 +784,976 @@ public void selectBestPoolsDoesNotFilterPoolsBySLAWhenConsumerHasOnlyNullEntitle
+        productWithRoleSatisfied.setAttribute("roles", "RHEL Workstation");
+        Pool poolThatSatisfiesRole = new Pool();
+        poolThatSatisfiesRole.setProduct(productWithRoleSatisfied);
+        Entitlement entitlementThatSatisfiesRole = new Entitlement();
+        entitlementThatSatisfiesRole.setPool(poolThatSatisfiesRole);
+        compliance.addCompliantProduct("compliant-product1", entitlementThatSatisfiesRole);
+
+        // Candidate pools:
+        Product prodRH00009 = new Product();
+        prodRH00009.setAttribute("roles", "RHEL Server");
+        prodRH00009.setAttribute("addons", "Smart Management");
+        Pool RH00009 = TestUtil.createPool(owner, prodRH00009);
+        RH00009.setId("RH00009");
+
+        Product prodMCT0352 = new Product();
+        prodMCT0352.setAttribute("roles", "RHEL Workstation");
+        Pool MCT0352 = TestUtil.createPool(owner, prodMCT0352);
+        MCT0352.setId("MCT0352");
+
+        Product prodRH00030 = new Product();
+        prodRH00030.setAttribute("addons", "RHEL EUS");
+        Pool RH00030 = TestUtil.createPool(owner, prodRH00030);
+        RH00030.setId("RH00030");
+
+        jsRules.reinitTo("test_name_space");
+        JsonJsContext args = new JsonJsContext(mapper);
+        args.put("log", log, false);
+        args.put("consumer", consumer);
+        args.put("compliance", compliance);
+
+        args.put("pool", RH00009);
+        Double RH00009Priority = jsRules.invokeMethod("get_pool_priority_test", args);
+
+        args.put("pool", MCT0352);
+        Double MCT0352Priority = jsRules.invokeMethod("get_pool_priority_test", args);
+
+        args.put("pool", RH00030);
+        Double RH00030Priority = jsRules.invokeMethod("get_pool_priority_test", args);
+
+        assertTrue("Pool RH00030 should have a higher priority than pool MCT0352.",
+            RH00030Priority > MCT0352Priority);
+
+        assertTrue("Pool MCT0352 should have a higher priority than pool RH00009.",
+            MCT0352Priority > RH00009Priority);
+    }
+
+    @SuppressWarnings("checkstyle:localvariablename")
+    @Test
+    public void testSysPurposePoolPriorityUseCase9PremiumSLADevelopmentUsage() throws NoSuchMethodException {
+        // Consumer specified syspurpose attributes:
+        consumer.setRole("RHEL Server");
+        consumer.setServiceLevel("Premium");
+        consumer.setUsage("Development");
+
+        // --- No satisfied syspurpose attributes on the consumer ---
+
+        // Candidate pools:
+        Product prodRH00009 = new Product();
+        prodRH00009.setAttribute("roles", "RHEL Server");
+        prodRH00009.setAttribute("addons", "Smart Management");
+        prodRH00009.setAttribute("support_level", "Standard");
+        prodRH00009.setAttribute("usage", "Production");
+        Pool RH00009 = TestUtil.createPool(owner, prodRH00009);
+        RH00009.setId("RH00009");
+
+        Product prodRH00008 = new Product();
+        prodRH00008.setAttribute("roles", "RHEL Server");
```

Comment:
All these calls to `setAttribute` really increase the size of the tests.  How would you feel about implementing a private method like:

```java
private Product createSysPurposeProduct(String id, String[] roles, String[] addons, String supportLevel, String usage) {
    Product p = new Product();
    p.setId(id);
    if (roles != null) {
        p.setAttribute("roles", String.join(", ", roles));    // or whatever the idiomatic Java for joining an array together is
    }
    ...
}
```

Reply:
I'm creating this method, but I think it would be better if it accepts just a cvs String for roles and addons, which we can pass csv-formatted strings directly, because when reading the tests, this:
```
Product prod1 = createSysPurposeProduct(null, "RHEL Server,RHEL Workstation", "RHEL EUS", null, "Production");
```
is more readable than this (for me):
```
Product prod1 = createSysPurposeProduct(null, new String[]{"RHEL Server", "RHEL Workstation"}, new String[]{"RHEL EUS"}, null, "Production");
```
especially since intellij will hint the argument names of strings when you pass them in, but not when you pass in arrays of strings.

Reply:
Yeah, a comma-delimited string works for me.  In fact, I think it's probably better since we're storing those values as comma-delimited strings in the DB so that keeps the method more in-line with what's happening elsewhere.  Good idea.